lib/stdlib/src/binary.erl

```diff
@@ -684,34 +780,35 @@ split(_, _, _) ->
       Replacement :: binary() | fun((binary()) -> binary()),
       Result :: binary().
 
-replace(H,N,R) ->
+replace(H, N, R) ->
     try
-        replace(H,N,R,[])
+        replace(H, N, R, [])
     catch
         error:Reason ->
             error_with_info(Reason, [H,N,R])
     end.
 
 -doc """
-Constructs a new binary by replacing the parts in `Subject` matching `Pattern`
-with `Replacement` if given as a literal `t:binary/0` or with the result of
-applying `Replacement` to a matching subpart if given as a `fun`.
+Constructs a new binary by replacing the parts in `Subject` matching
+`Pattern` with `Replacement` if given as a literal
+[`binary()`](`t:binary/0`) or with the result of applying
+`Replacement` to a matching subpart if given as a `fun`.
 
-If `Replacement` is given as a `t:binary/0` and the matching subpart of
+If `Replacement` is given as a [`binary()`](`t:binary/0`) and the matching subpart of
```

Comment:
When deciding how ExDoc should parse code segments for auto-links, I decided that we should not include `t:binary()` in the parsing, because we would want to write them as `t:binary/0`. The idea is that using `binary/0` makes it look the same no matter if it is 0 or 1 arity, while the `binary()` notation is only possible for 0-arity types.

We should either way write a style guide into the documentation howto about how to refer to types, function, callbacks and modules (unless there already is one that I did not see).

Reply:
> We should either way write a style guide into the documentation howto about how to refer to types, function, callbacks and modules (unless there already is one that I did not see).

https://www.erlang.org/doc/system/documentation.html#links-in-markdown

Reply:
I find the type `binary/0` confusing because it could also refer to the function `binary/0`. I prefer `binary()` which is the syntax used in specs. Also, I think that syntax would be more familiar to the average user.

A type of arity one I would like to write as `gb_sets:set(Element)`.

I agree that we should agree on how to refer to types and add that to the style guide.

Reply:
> I find the type binary/0 confusing because it could also refer to the function binary/0. I prefer binary() which is the syntax used in specs. Also, I think that syntax would be more familiar to the average user.

I think that if it is a type or function is often very obvious by the context in which it is used. When it is not obvious it can be made so by writing "the type `t:binary/0`" (as you did in your example). IMO having consistency in how we refer to things (functions, types, callbacks, ...) is more important, so that when reading the docs things look and feel uniform.

Reply:
I've changed back to use `t:binary/0`.